e2e/common/utils.go

```diff
@@ -0,0 +1,194 @@
+	}
+	return nil
+}
+
+func UninstallKubeClarity() error {
+	// uninstall kubeclarity
+	cmd := exec.Command("helm", "uninstall", KubeClarityHelmReleaseName, "-n", KubeClarityNamespace)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to uninstall kubeclarity: %s. %v", out, err)
+	}
+
+	cmd2 := exec.Command("kubectl", "-n", KubeClarityNamespace, "delete", "pvc", "data-kubeclarity-kubeclarity-postgresql-0")
+	out, err = cmd2.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to delete pvc: %s. %v", out, err)
+	}
+
+	cmd3 := exec.Command("kubectl", "delete", "ns", KubeClarityNamespace)
+	out, err = cmd3.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to delete kubeclarity namespace: %s. %v", out, err)
+	}
+
+	return nil
+}
+
+func PortForwardToKubeClarity(stopCh chan struct{}) {
+	go func() {
+		err := portForward("service", KubeClarityNamespace, KubeClarityServiceName, KubeClarityPortForwardHostPort, KubeClarityPortForwardTargetPort, stopCh)
+		if err != nil {
+			println("port forward failed. %v", err)
+			return
+		}
+	}()
+	time.Sleep(3 * time.Second)
+}
+
+func StringPtr(val string) *string {
+	ret := val
+	return &ret
+}
+
+//TODO use https://github.com/kubernetes-sigs/e2e-framework/tree/main/examples/wait_for_resources
+func WaitForPodRunning(client klient.Client, ns string, labelSelector string) error {
+	podList := v1.PodList{}
+	ticker := time.NewTicker(3 * time.Second)
+	defer ticker.Stop()
+	timeout := time.NewTimer(3 * time.Minute)
+	for {
+		select {
+		case <-timeout.C:
+			return fmt.Errorf("timeout reached")
+		case <-ticker.C:
+			if err := client.Resources(ns).List(context.TODO(), &podList, func(lo *v12.ListOptions) {
+				lo.LabelSelector = labelSelector
+			}); err != nil {
+				return fmt.Errorf("failed to get pod in namespace: %v and labels: %v. %v", ns, labelSelector, err)
+			}
+			pod := podList.Items[0]
+			if pod.Status.Phase == v1.PodRunning {
+				return nil
+			}
+		}
+	}
+}
+
+func CreateNamespace(client klient.Client ,name string) error {
+	var ns = v1.Namespace{
+		TypeMeta:   v12.TypeMeta{
+			Kind:       "Namespace",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v12.ObjectMeta{
+			Name:                       name,
+		},
+	}
+	if err := client.Resources(name).Create(context.TODO(), &ns); err != nil {
+		return err
+	}
+	return nil
+}
+
+// NON EXPORTED:
+
+func portForward(kind, namespace, name, hostPort, targetPort string, stopCh chan struct{}) error {
+	cmd := exec.Command("kubectl", "port-forward", "-n", namespace,
+		fmt.Sprintf("%s/%s", kind, name), fmt.Sprintf("%s:%s", hostPort, targetPort))
+
+	processExitedCh := make(chan struct{})
+	var output []byte
+	var err error
+	go func() {
+		output, err = cmd.CombinedOutput()
+		if err != nil {
+			processExitedCh <- struct{}{}
+		}
+	}()
+
+	select {
+	case <-stopCh:
+		if err := cmd.Process.Kill(); err != nil {
+			return fmt.Errorf("failed to kill process: %v", err)
```

Comment:
do we care? maybe just a print and `return nil` is enough?

Reply:
There is no harm in printing this error. It does not fail the test.

Reply:
In general it seems odd to me to return an error of a close/kill failure - there is nothing to do with it.